hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/x509/certificate/authority/TestDefaultCAServer.java

```diff
@@ -197,11 +200,14 @@ public void testRequestCertificate() throws IOException,
     testCA.init(new SecurityConfig(conf),
         SELF_SIGNED_CA);
 
-    Future<X509CertificateHolder> holder = testCA.requestCertificate(csrString,
-        CertificateApprover.ApprovalType.TESTING_AUTOMATIC, SCM);
+    Future<CertPath> holder = testCA.requestCertificate(
+        csrString, CertificateApprover.ApprovalType.TESTING_AUTOMATIC, SCM);
     // Right now our calls are synchronous. Eventually this will have to wait.
     assertTrue(holder.isDone());
-    assertNotNull(holder.get());
+    //Test that the cert path returned contains the CA certificate in proper
+    // place
+    assertEquals(holder.get().getCertificates().get(1),
```

Comment:
Shouldn't we assert as well that the returned certificate at the first place matches the data that we specified in the CSR?

Reply:
This is a repeated comment, see my answer in the other one.